Foundation/testsuite/src/PathTest.cpp

```diff
@@ -1619,6 +1619,28 @@ void PathTest::testWindowsSystem()
 #endif
 }
 
+void PathTest::testSelf()
+{
+	std::string self = Path::self();
+	std::cout << self << std::endl;
+#if (defined(POCO_OS_FAMILY_WINDOWS) && !defined(_WIN32_WCE)) \
+	|| POCO_OS == POCO_OS_MAC_OS_X                            \
+	|| POCO_OS == POCO_OS_FREE_BSD                            \
+	|| POCO_OS == POCO_OS_SOLARIS                             \
+	|| POCO_OS == POCO_OS_LINUX                               \
+	|| POCO_OS == POCO_OS_ANDROID
+
+	assertTrue(!self.empty());
+	Path p(self);
+	assertTrue(p.isAbsolute());
+	assertTrue(p.isFile());
+	assertTrue(self.find("testrunner") != std::string::npos);
```

Comment:
this test must somehow be skipped on windows because the executable is named `TestSuite` there

Reply:
@aleks-f I skipped and assertTrue on Windows. Or you mean skip `testSelf` on Windows?

Reply:
@JackyWoo windows should be tested